chain/network/src/network_protocol/tests.rs

```diff
@@ -124,7 +124,7 @@ fn serialize_deserialize() -> anyhow::Result<()> {
         PeerMessage::Transaction(data::make_signed_transaction(&mut rng)),
         PeerMessage::Routed(routed_message1),
         PeerMessage::Routed(routed_message2),
-        PeerMessage::Disconnect(Disconnect { allow_reconnect: true }),
+        PeerMessage::Disconnect(Disconnect { remove_from_recent_outbound_connections: false }),
```

Comment:
FYI remember that if a proto is missing a field, the default value is used ("false" for boolean fields). If you want to distinguish false from unspecified, you need to wrap the bool into a proto message (then a missing message means "unspecified" which is different that a message with "false" inside).

Please consider what is the expected behavior in case of receiving a Disconnect message from neard with an older version, and document that in the PR description.

Reply:
In the absence of an explicit signal on whether the disconnect was accompanied by some reason to stay disconnected (ban, etc.), attempting to reconnect seems like the correct behavior.